taichi/ir/state_machine.cpp

```diff
@@ -0,0 +1,331 @@
+#include "state_machine.h"
+
+TLANG_NAMESPACE_BEGIN
+
+std::unique_ptr<std::unordered_set<AtomicOpStmt *>> StateMachine::used_atomics;
```

Comment:
Shall we make it a member instead of a global var? (If this is too troublesome we should leave this refactoring to a future PR.)

Reply:
This is the default initialization of the static member variable... Is this statement unnecessary?

Reply:
You are right - I missed `StateMachine::`. Sorry about that.

Reply:
Wait, I feel like the ownership of this static variable is still unclear. To me, it seems that `StateMachine` should not own this. Its owner should be `VariableOptimization`.

Reply:
It's more convenient to write it in `StateMachine` now, because I do all modification on IR in `StateMachine`......